packages/ketcher-core/src/domain/serializers/ket/schema.json

```diff
@@ -713,6 +758,270 @@
+        },
+        "templateId": {
+          "type": "string"
+        },
+        "alias": {
+          "type": "string"
+        },
+        "position": {
+          "type": "object",
+          "required": ["x", "y"],
+          "properties": {
+            "x": {
+              "type": "number"
+            },
+            "y": {
+              "type": "number"
+            }
+          }
+        }
+      }
+    },
+    "group": {
+      "type": "object",
+      "required": ["type", "groupClass", "nodes"],
+      "properties": {
+        "type": {
+          "const": "group"
+        },
+        "groupClass": {
+          "type": "string",
+          "enum": ["Generic", "Variant", "RepeatUnit"]
+        },
+        "attachmentPoints": {
+          "type": "array",
+          "minItems": 0,
+          "items": {
+            "type": "object",
+            "required": ["id", "variantList"],
+            "properties": {
+              "id": {
+                "type": "string"
+              },
+              "variantList": {
+                "allOf": [
+                  {
+                    "$ref": "#/definitions/group"
+                  },
+                  {
+                    "properties": {
+                      "groupClass": {
+                        "const": "Variant"
+                      }
+                    },
+                    "required": ["groupClass"]
+                  }
+                ]
+              }
+            }
+          }
+        },
+        "nodes": {
+          "type": "array",
+          "items": {
+            "type": "object",
+            "required": ["$ref"],
+            "properties": {
+              "$ref": {
+                "type": "string",
+                "pattern": "^(group\\d+|monomer\\d+|monomerTemplate-.*)"
+              }
+            }
+          }
+        }
+      },
+      "patternProperties": {
+        "^monomer\\d+": {
+          "$ref": "#/definitions/monomer"
+        },
+        "^group\\d+": {
+          "$ref": "#/definitions/group"
+        },
+        "^monomerTemplate-.*": {
+          "$ref": "#/definitions/monomerTemplate"
+        }
+      },
+      "allOf": [
+        {
+          "if": {
+            "properties": {
+              "groupClass": {
+                "const": "Generic"
+              }
+            }
+          },
+          "then": {
+            "properties": {
+              "connections": {
+                "type": "array",
+                "minItems": 0,
+                "items": {
+                  "$ref": "#/definitions/nodesConnection"
+                }
+              }
+            }
+          }
+        },
+        {
+          "if": {
+            "properties": {
+              "groupClass": {
+                "const": "RepeatUnit"
+              }
+            }
+          },
+          "then": {
+            "properties": {
+              "repeatRange": {
+                "type": "object",
+                "required": ["from"],
```

Comment:
I believe it will not work, because if/then properties repeates many times on the same level of json. JS during parsing will ignore all of them  except the last ones. I believe it should be written like this:
<img width="333" alt="image" src="https://github.com/epam/ketcher/assets/14859362/e4071267-4d3d-45e5-9156-236408641ac6">